src/server/server_family.cc

```diff
@@ -3902,6 +3903,11 @@ void ServerFamily::ReplConf(CmdArgList args, const CommandContext& cmd_cntx) {
       if (info) {
         info->id = arg;
       }
+      // If we tried to replicate from ourself reply with an error
+      if (arg == master_replid_) {
```

Comment:
hmm, why here, why not in replica inside `HandleCapaDflyResp` ?
one of the reasons why it is preferable is that during updates, the (old) master does not have this fix, so we will still have the infinite loop. `HandleCapaDflyResp` is on replica side so you propagate a good behaviour naturally.